### ISSUE TITLE
storing selections

### ISSUE THREAD

**Issue by ehesch** [Metropolitan-Council/regionalparks.acs#13]
Hi @eroten - thanks for your help today! The structure is starting to make a lot more sense! I've created a series of `mod_summary_*` files which will be passed to the "ACS summary" tab. Still, I'm having some trouble saving the user-selections (`mod_summary_selections.R`) and then passing them to subsequent modules. 

I've been trying to take the user selections and pass them to `mod_summary_table.R` to test if the filtering is working. My list of user-selected inputs seems to be the problem (empty values + bad structure).

First the user selects inputs within `mod_summary_selections_server` 
```r 
return(
    list(
      select_acs <- reactive({input$ACS}),
      select_agency <- reactive({input$agency}),
      select_distance <- reactive({input$distance}),
      select_type <- reactive({input$type}),
      select_status <- reactive({input$status})
    )
  )
```
In the `app_server` the user-selected values are assigned to `selected_vars`
```r
selected_vars <- mod_summary_selections_ui("summary_selections_ui_1")
```

And then the `mod_summary_table_server` accepts the selected values as an input (along with the raw data)
```r
callModule(mod_summary_table_server, "summary_table_ui_1", selected_vars, long_buffer_data)
```

But then a `browser()` shows that the `selected_vars()` is not behaving like it needs to within the `mod_summary_table_server`:
```r
  output$datatable <- renderDataTable({
    long_buffer_data  %>%
      filter(type == selected_vars$select_type())
  })
```

(Happy to hop on a quick call tomorrow if that is easier!)

**Comment by eroten** [comment]
Hi @ehesch!

Just looking at what you've written in this issue, I have an idea of what is going on. 

Returning a regular `list()` will cause issues with Shiny.  A `list()` is a static object, which means it will not update when one of its values changes. Reactive values need to be stored as `reactiveValues()`, which is essentially a :sparkles: fancy :sparkles: list. You also don't need to fill the `reactiveValues` when its created; it can be empty and filled later. 

Here is how I would suggest handling this in `mod_summary_selections_server.R`

```r
input_values <- reactiveValues() # start with an empty reactiveValues object. 

observeEvent(input$ACS, { # only update when the user changes the ACS input
input_values$ACS <- input$ACS # create/update the ACS input value in our reactiveValues object
})

observeEvent(input$agency, {
input_values$agency <- input$agency
})

# ... repeat this pattern for each input

return(input_values) # return the entire reactiveValues object
```

Then, over in `app_server.R`, we assign this return as `selected_vars`. When we access `selected_vars`, we can just access the values like a list, rather than as a `reactive()` function. 

I dug up my favorite training for actually figuring out reactivity and the difference between `reactive()` and `observe()`. This training was a serious game-changer, and it also includes one of my favorite haikus of all time. There are two parts, and if you follow along with the exercises (which I 100% recommend, as Cheng will show solutions and not-solutions and explain 🎵 why 🎵 ) it will take you ~2.5 hours. I also paused the video a lot to write notes.

The talks are [here](https://rstudio.com/resources/shiny-dev-con/reactivity-pt-1-joe-cheng/) and [here](https://rstudio.com/resources/shiny-dev-con/reactivity-pt-2/), and the GitHub repo with the presentation and exercises is [here](https://github.com/rstudio/ShinyDeveloperConference/tree/master/Reactivity). Go forth and learn!

**Comment by eroten** [comment]
I made these changes and a few others this morning on the [reactivity-fix](https://github.com/Metropolitan-Council/regionalparks.acs/tree/reactivity-fix) branch, making detailed commit messages as I went. The app will run and the table displayed in the summary tab is responds to changes in the summary inputs 💃 

To summarize, I implemented the change I suggested in the last comment, changed a few IDs to be more consistent and descriptive, and started `mod_summary_utils.R`. This module has no UI elements, and is our "calculator" for taking the summary inputs and making the data table that will be used in `mod_summary_table.R`. I deleted 'mod_summary_filtered.R'.

I also worked on getting a default value to populate on the ACS map tab to no avail. I'll give it another go next week.
